### PR TITLE
Improved Modular Arithmetic Module

### PR DESCRIPTION
**Pull request by SansPapyrus683**
Formatted code and added Python support. Not sure if I should've used the built-in `pow` function though.

### CODE REVIEW

File: content/4_Gold/Modular.mdx [hunk 1]
```diff
@@ -296,19 +316,19 @@ We can also use the extended Euclidean algorithm. See the module in the
...
 
 ```cpp
 int main() {
 	{
 		int a = 1e8, b = 1e8, c = 1e8;
-		cout << (ll)a*b%MOD*c%MOD << "\n"; // 49000000
+		cout << (ll) a * b % MOD * c % MOD << "\n"; // 49000000
 	}
 	{
 		mi a = 1e8, b = 1e8, c = 1e8;
```
Comment (maggie-j-liu) [inline content/4_Gold/Modular.mdx:328]: maybe the template could be added as a codesnip?

Comment (SansPapyrus683) [reply content/4_Gold/Modular.mdx:328]: You're not talking about the entire benq template right?

Comment (maggie-j-liu) [reply content/4_Gold/Modular.mdx:328]: just the modInt one?

**Comment by SansPapyrus683** [comment]
plz work

File: content/4_Gold/Modular.mdx [hunk 2]
```diff
@@ -296,19 +315,49 @@ We can also use the extended Euclidean algorithm. See the module in the
...
+	return a;
+}
+mi operator+(mi a, mi b) { return a += b; }
+mi operator-(mi a, mi b) { return a -= b; }
+mi operator*(mi a, mi b) { return mi((long long) a.v * b.v); }
+mi& operator*=(mi& a, mi b) { return a = a * b; }
+mi pow(mi a, long long p) {
+    assert(p >= 0);
+	return p == 0 ? 1 : pow(a * a, p / 2) * (p & 1 ? a : 1);
+}
```
Comment (dongliu0426) [inline content/4_Gold/Modular.mdx:347]:
```suggestion
mi pow(mi a, long long p) {
	assert(p >= 0);
	return p == 0 ? 1 : pow(a * a, p / 2) * (p & 1 ? a : 1);
}
```

**Review by maggie-j-liu — APPROVED** [review]

**Pull request merged**